lms/services/moodle.py

```diff
@@ -22,6 +23,23 @@ class Function(str, Enum):
     """Returns a list of pages in a provided list of courses."""
 
 
+class File(TypedDict):
+    """Represents a file or folder in an LMS's file storage."""
+
+    type: Literal["File", "Folder"]
+    mime_type: NotRequired[Literal["text/html", "application/pdf", "video"]]
+
+    id: str
+    """ID in our system"""
+    lms_id: str
+    """Raw ID in the LMS"""
+
+    display_name: str
+
+    updated_at: NotRequired[str | None]
```

Comment:
This doesn't fully match the type defined in TypeScript, where it's set as `updated_at?: string`. That means that either the property is not set, or it is set with a string.

If I'm not wrong, `NotRequired[str | None]` means that it can also be set as `null`. It probably wouldn't make a difference in the FE because of how we currently use it, but `undefined` (or "not set") and `null` are not 100% the same and could lead to some error in future.

Should we change the type definition in the FE?

Reply:
In a different PR we agreed to use `foo: NotRequired[type]` as the equivalent of `property?: type`. I think it probably makes sense to do the same here.

Reply:
Yup, changed to that :+1: